crates/cloudmqtt-core/src/client/mod.rs

```diff
@@ -97,17 +97,17 @@ where
 
 impl Default for MqttClientFSM<UsizePacketIdentifierStore> {
     fn default() -> Self {
-        Self::new(UsizePacketIdentifierStore::default())
+        Self::const_new(UsizePacketIdentifierStore::default())
     }
 }
 
 impl<CPIS> MqttClientFSM<CPIS>
 where
     CPIS: PacketIdentifierStore,
 {
-    pub fn new(client_pis: CPIS) -> MqttClientFSM<CPIS> {
+    pub const fn const_new(client_pis: CPIS) -> MqttClientFSM<CPIS> {
```

Comment:
You don't need to call it const_new, just make it const

Reply:
I kinda figured that this is more or less the convention, especially since there's a `Default` implementation, but yes, sure!

Reply:
I'd be happy to learn more about it! But I've only see `const_new` when it does something different than the usual `new`